staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -216,7 +215,9 @@ func (u *udsGRPCConnector) connect(_ context.Context) (proxier, error) {
 	// we cannot use ctx just for dialing and control the connection lifetime separately.
 	// See https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/357.
 	tunnelCtx := context.TODO()
-	tunnel, err := client.CreateSingleUseGrpcTunnel(tunnelCtx, udsName, dialOption,
+	tunnel, err := client.CreateSingleUseGrpcTunnelWithContext(connectCtx, tunnelCtx, udsName, dialOption,
+		grpc.WithBlock(),
```

Comment:
Since we are using `grpc.WithBlock()`, I think I'm in favor of setting `grpc.WithTimeout()` with seems to accomplish the same thing, but would clean up some of the changes needed in this file:

https://github.com/kubernetes/kubernetes/blob/master/vendor/google.golang.org/grpc/clientconn.go#L217-L220

Reply:
I don't feel too strongly about this since I'm not too familiar with the context being passed into `connect()`, I'll defer to you and other reviewers on this but just my two cents

Reply:
I'm not sure I understand.

Passing the stricter 10s timeout via grpc.WithTimeout() would be redundant with the ctx we pass to grpc.DialContext(), since it comes from connectCtx.

Reply:
> Passing the stricter 10s timeout via grpc.WithTimeout() would be redundant with the ctx we pass to grpc.DialContext(), since it comes from connectCtx.

I meant to set `grpc.WithTimeout()` in replacement of setting a context timeout in `connectCtx`, which would also remove the diff of needing to separate requestCtx and connectCtx. 

Another reason would be that setting the context timeout from `createDialer` would apply for all egress protocols, not just gRPC.